### PR TITLE
chassisd: Fix crash on exit on linecard

### PR DESCRIPTION
**Pull request by patrickmacarthur**
#### Description
Set the `config_manager` variable to `None` if we are running on a linecard and thus don't need to set up the config manager.

#### Motivation and Context
During cleanup, the chassid service tries to clean up the `config_manager`, but the `config_manager` variable is only ever initialized if we are on the supervisor. Thus, checking if it is `None` is insufficient because this results in an `UnboundLocalError` that prevents the cleanup from succeeding on a linecard.

#### How Has This Been Tested?
This was tested via the sonic-mgmt `platform_tests/daemon/test_chassisd.py` test. Without this change, the `test_pmon_chassisd_stop_and_start_status` test would fail and a traceback would be present in syslog. The test passes with this patch applied.

### CODE REVIEW

**Review by wenyiz2021 — COMMENTED** [review]
@patrickmacarthur thanks for the fix, can you please add a UT to cover this? PR is blocked by coverage check

**Comment by abdosi** [comment]
@gechiang for viz.

**Review by abdosi — DISMISSED** [review]

**Comment by gechiang** [comment]
@patrickmacarthur can you also include a test to cover the one line change you fixed?  This is required before the change can be merged.
You can reference this original PR to see how the testcases were originally added and add one test to cover your new change for non-supervisor case.
https://github.com/sonic-net/sonic-platform-daemons/pull/97

**Comment by abdosi** [comment]
@bmridul

**Review by gechiang — APPROVED** [review]
LGTM

**Comment by StormLiangMS** [comment]
@patrickmacarthur cherry pick to 202211 with conflict, could you submit PR separately for 202211? 
@gechiang  @abdosi for vis.

**Pull request merged**
